File: 02.Array_problems/BogoSort.java
package Array_problems;

import java.util.*;

/**
 * BogoSort
 */


public class BogoSort {

    public static boolean isSorted(int[] arr){
        for(int i=1; i<arr.length; i++){
            if(arr[i-1] > arr[i]) return false;
        }
        return true;
    }

    public static void shuffleArray(int[] arr){
        Random rand = new Random();
        for(int i=arr.length-1; i>0; i--){
            int j = rand.nextInt(i+1);
            int temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }
    }

    public static void bogoSort(int[] arr){
        while (!isSorted(arr)) {
            shuffleArray(arr);
        }
    }

    public static void main(String[] args) {
        int[] arr = {4,2,5,1,7,3};
        System.out.println("Unsorted Array:"+ Arrays.toString(arr));
        bogoSort(arr);
        System.out.println("Sorted Array:"+ Arrays.toString(arr));
    }
}

File: 02.Array_problems/BogoSort.js

// Second Worst Sorting Algorithm - Bogo Sort

function isSorted(arr){
    for(let i=1; i<arr.length; i++){
        if(arr[i-1] > arr[i]) return false;
    }
    return true;
}

function bogoSort(arr){
    // check till the array is sorted
    while(!isSorted(arr)){
        // shuffle the array
        for(let i=arr.length-1; i>0; i--){
            let j = Math.floor(Math.random() * (i+1));
            [arr[i], arr[j]] = [arr[j], arr[i]];
        }
    }
    return arr;
}

let arr = [3,2,5,1,6,7];
console.log("Unsorted array: ",arr);

let res = bogoSort(arr);
console.log("Sorted array:", res);